db/migrate/20160322205357_remove_host_id_from_switch.rb

```diff
@@ -0,0 +1,41 @@
+class RemoveHostIdFromSwitch < ActiveRecord::Migration[5.0]
+  class Host < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled # disable STI
+    has_many :host_switches,
+             :dependent   => :destroy,
+             :class_name  => 'RemoveHostIdFromSwitch::HostSwitch'
+    has_many :switches,
+             :through => :host_switches,
+             :class_name  => 'RemoveHostIdFromSwitch::Switch'
+  end
+
+  class Switch < ActiveRecord::Base
+    has_many :host_switches,
+             :dependent   => :destroy,
+             :class_name  => 'RemoveHostIdFromSwitch::HostSwitch'
+    has_many :hosts,
+             :through => :host_switches,
+             :class_name => 'RemoveHostIdFromSwitch::Host'
+  end
+
+  class HostSwitch < ActiveRecord::Base
+    belongs_to :host,
+               :class_name  => 'RemoveHostIdFromSwitch::Host'
+    belongs_to :switch,
+               :class_name => 'RemoveHostIdFromSwitch::Switch'
+  end
+
+  def up
+    remove_column :switches, :host_id, :bigint
+  end
+
+  def down
+    add_column :switches, :host_id, :bigint
+    say_with_time('Populating switches.host_id from host_switches table') do
+      Switch.all.each do |switch|
+        switch.host_id = switch.hosts[0].id
```

Comment:
Hmmm... good point. If this is run after DVS are in the inventory, then this down migration for DVS won't be well defined. Shall we drop a DVS from the inventory in this down migration? Or simply assign the first host to a DVS (as the code is now)?